docker/Makefile

```
@@ -148,7 +148,7 @@ venv: requirements.txt
 	pyenv install --skip-existing $(PY_VERSION) &&\
 	pyenv local $(PY_VERSION) &&\
 	python3 -mvenv venv && \
-	pip install -r requirements.txt &&\
+	for i in {1..5}; do pip install -r requirements.txt && break || sleep 5; done &&\
```

Will the 5th failure, followed by a successful sleep, have a success status and fail to break the chain?

---

That's a good point.. the `sleep 5` could be replaced by something like `(sleep 1; if [[ $i -eq 5 ]]; then return 1; fi)`, but we need to get that "1" out of the for loop.something like `var = ; (for loop; ... var = 1; done); return var`

---

```suggestion
	for i in {1..5}; do pip install -r requirements.txt && break || (sleep 5; false); done &&\
```

OR using our `i` for an linear backoff:

```suggestion
	for i in {0..5}; do sleep i; pip install -r requirements.txt && break; done &&\
```